src/leap/mail/imap/messages.py

```diff
@@ -216,62 +64,30 @@ def getFlags(self):
         :return: The flags, represented as strings
         :rtype: tuple
         """
-        uid = self._uid
-
-        flags = set([])
-        fdoc = self.fdoc
-        if fdoc:
-            flags = set(fdoc.content.get(self.FLAGS_KEY, None))
-
-        msgcol = self._collection
-
-        # We treat the recent flag specially: gotten from
-        # a mailbox-level document.
-        if msgcol and uid in msgcol.recent_flags:
-            flags.add(fields.RECENT_FLAG)
-        if flags:
-            flags = map(str, flags)
-        return tuple(flags)
+        return self.message.get_flags()
 
     # setFlags not in the interface spec but we use it with store command.
 
-    def setFlags(self, flags, mode):
-        """
-        Sets the flags for this message
+    # XXX if we can move it to a collection method, we don't need to pass
+    # collection to the IMAPMessage
 
-        :param flags: the flags to update in the message.
-        :type flags: tuple of str
-        :param mode: the mode for setting. 1 is append, -1 is remove, 0 set.
-        :type mode: int
-        """
-        leap_assert(isinstance(flags, tuple), "flags need to be a tuple")
-        mbox, uid = self._mbox, self._uid
-
-        APPEND = 1
-        REMOVE = -1
-        SET = 0
-
-        with fdoc_locks[mbox][uid]:
-            doc = self.fdoc
-            if not doc:
-                logger.warning(
-                    "Could not find FDOC for %r:%s while setting flags!" %
-                    (mbox, uid))
-                return
-            current = doc.content[self.FLAGS_KEY]
-            if mode == APPEND:
-                newflags = tuple(set(tuple(current) + flags))
-            elif mode == REMOVE:
-                newflags = tuple(set(current).difference(set(flags)))
-            elif mode == SET:
-                newflags = flags
-            new_fdoc = {
-                self.FLAGS_KEY: newflags,
-                self.SEEN_KEY: self.SEEN_FLAG in newflags,
-                self.DEL_KEY: self.DELETED_FLAG in newflags}
-            self._collection.memstore.update_flags(mbox, uid, new_fdoc)
-
-        return map(str, newflags)
+    # lookup method? IMAPMailbox?
+
+    #def setFlags(self, flags, mode):
+        #"""
+        #Sets the flags for this message
+#
+        #:param flags: the flags to update in the message.
+        #:type flags: tuple of str
+        #:param mode: the mode for setting. 1 is append, -1 is remove, 0 set.
+        #:type mode: int
+        #"""
+        #leap_assert(isinstance(flags, tuple), "flags need to be a tuple")
+        # XXX
+        # return new flags
+        # map to str
+        #self.message.set_flags(flags, mode)
+        #self.collection.update_flags(self.message, flags, mode)
```

Comment:
Is this waiting for collection to implement setFlags? Or can it already be deleted? I see this function is used, and will fail in `IMAPMessageCollection.set_flags`.